avocado_vt/test.py

```diff
@@ -217,28 +218,48 @@ def setUp(self):
         Avocado-vt uses custom setUp/test/tearDown handling and unlike
         Avocado it allows skipping tests from any phase. To convince
         Avocado to allow skips let's say our tests run during setUp
-        phase and don't do anything during runTest/tearDown.
+        phase and report the status in test.
         """
         env_lang = os.environ.get('LANG')
         os.environ['LANG'] = 'C'
         try:
             self._runTest()
+            self.__status = "PASS"
         # This trick will give better reporting of virt tests being executed
         # into avocado (skips, warns and errors will display correctly)
-        except error.TestNAError, details:
-            raise exceptions.TestSkipError(details)
-        except error.TestWarn, details:
-            raise exceptions.TestWarn(details)
-        except error.TestError, details:
-            raise exceptions.TestError(details)
-        except error.TestFail, details:
-            raise exceptions.TestFail(details)
+        except:  # Old-style exceptions are not inherited from Exception()
```

Comment:
This will also catch TestSkipError, and lead to a TestError as we raise it in runTest